support/ray_cluster_client.go

```diff
@@ -45,20 +46,47 @@ type RayJobLogsResponse struct {
 	Logs string `json:"logs"`
 }
 
+type RayClusterClientConfig struct {
+	Address             string
+	Client              *http.Client
+	SkipTlsVerification bool
+}
+
 var _ RayClusterClient = (*rayClusterClient)(nil)
 
 type rayClusterClient struct {
-	endpoint url.URL
+	endpoint    url.URL
+	httpClient  *http.Client
+	bearerToken string
 }
 
 type RayClusterClient interface {
 	CreateJob(job *RayJobSetup) (*RayJobResponse, error)
 	GetJobDetails(jobID string) (*RayJobDetailsResponse, error)
 	GetJobLogs(jobID string) (string, error)
+	GetJobs() ([]map[string]interface{}, error)
 }
 
-func NewRayClusterClient(dashboardEndpoint url.URL) RayClusterClient {
-	return &rayClusterClient{endpoint: dashboardEndpoint}
+var rayClusterApiClient RayClusterClient
+
+func NewRayClusterClient(config RayClusterClientConfig, bearerToken string) (RayClusterClient, error) {
```

Comment:
```suggestion
func GetRayClusterClient(config RayClusterClientConfig, bearerToken string) (RayClusterClient, error) {
```
Because you don't create a new client if it is already created.